jenkins/debian/debos/scripts/crush.sh

```diff
@@ -6,151 +6,181 @@ set -e
 
-# systemd hw database manager (connman is in use)
-find usr etc -name '*systemd-hwdb*' -prune -exec rm -r {} \;
+    # No bash, no need for completions
+    rm -rf usr/share/bash-completion
 
-# No need for fuse
-find usr etc -name '*fuse*' -prune -exec rm -r {} \;
+    # No zsh, no need for comletions
+    rm -rf usr/share/zsh/vendor-completions
 
-# lsb init function leftovers
-rm -rf usr/lib/lsb
+    # drop gcc-6 python helpers
+    rm -rf usr/share/gcc-6
 
-# Utils using ncurses
-rm -f usr/bin/pg
-rm -f usr/bin/watch
-rm -f usr/bin/slabtop
+    # Drop sysvinit leftovers
+    rm -rf etc/init.d
+    rm -rf etc/rc[0-6S].d
 
-# boot analyser
-rm -f usr/bin/systemd-analyze
+    # Drop upstart helpers
+    rm -rf etc/init
 
-# Only needed when adding libraries
-rm -f usr/sbin/ldconfig*
+    # Various xtables helpers
+    rm -rf usr/lib/xtables
 
-# Games, unused
-rm -rf usr/games
+    # Drop all locales
+    # TODO: only remaining locale is actually "C". Should we really remove it?
+    rm -rf usr/lib/locale/*
 
-# Unused systemd generators
-rm -f lib/systemd/system-generators/systemd-cryptsetup-generator
-rm -f lib/systemd/system-generators/systemd-debug-generator
-rm -f lib/systemd/system-generators/systemd-gpt-auto-generator
-rm -f lib/systemd/system-generators/systemd-hibernate-resume-generator
-rm -f lib/systemd/system-generators/systemd-rc-local-generator
-rm -f lib/systemd/system-generators/systemd-system-update-generator
-rm -f lib/systemd/system-generators/systemd-sysv-generator
+    # local compiler
+    rm -f usr/bin/localedef
 
-# Efi blobs
-rm -rf usr/lib/systemd/boot
+    # lsb init function leftovers
+    rm -rf usr/lib/lsb
 
-# Translation catalogs
-rm -rf usr/lib/systemd/catalog
+    # boot analyser
+    rm -f usr/bin/systemd-analyze
 
-# Misc systemd utils
-rm -f usr/bin/bootctl
-rm -f usr/bin/busctl
-rm -f usr/bin/hostnamectl
-rm -f usr/bin/localectl
-rm -f usr/bin/systemd-cat
-rm -f usr/bin/systemd-cgls
-rm -f usr/bin/systemd-cgtop
-rm -f usr/bin/systemd-delta
-rm -f usr/bin/systemd-detect-virt
-rm -f usr/bin/systemd-mount
-rm -f usr/bin/systemd-path
-rm -f usr/bin/systemd-run
-rm -f usr/bin/systemd-socket-activate
-
-# Remove pam module to authenticate against a DB
-# plus libdb-5.3.so that is only used by this pam module
-rm -f usr/lib/*/security/pam_userdb.so
-rm -f usr/lib/*/libdb-5.3.so
-
-# remove NSS support for nis, nisplus and hesiod
-rm -f usr/lib/*/libnss_hesiod*
-rm -f usr/lib/*/libnss_nis*
+    # Only needed when adding libraries
+    rm -f usr/sbin/ldconfig*
+
+    # Games, unused
+    rm -rf usr/games
+
+    # Remove pam module to authenticate against a DB
+    # plus libdb-5.3.so that is only used by this pam module
+    rm -f usr/lib/*/security/pam_userdb.so
+    rm -f usr/lib/*/libdb-5.3.so
+
+    # remove NSS support for nis, nisplus and hesiod
+    rm -f usr/lib/*/libnss_hesiod*
+    rm -f usr/lib/*/libnss_nis*
+}
+
+# Removing unneeded packages
+remove_packages()  {
+   UNNEEDED_PACKAGES="$@"
```

Comment:
I think the scope of `UNNEEDED_PACKAGES` is not local to this function, only `bash` provides support for this using `local`.  So it's probably better to either use a different name or directly use `$@` in the for-loop below.